frontend/src/metabase/components/UserAvatar/UserAvatar.tsx

```diff
@@ -0,0 +1,66 @@
+/* eslint-disable react/prop-types */
+import React from "react";
+
+import MetabaseUtils from "metabase/lib/utils";
+import { Avatar, AvatarProps } from "./UserAvatar.styled";
+
+interface UserAvatarProps extends AvatarProps {
+  user: User;
+}
+
+interface GroupProps {
+  user: Group;
+}
+
+interface User {
+  first_name: string | null;
+  last_name: string | null;
+  common_name: string;
+  email?: string;
+}
+
+interface Group {
+  first_name: string;
+}
+
+export default function UserAvatar({
+  user,
+  ...props
+}: UserAvatarProps | GroupProps) {
+  return <Avatar {...props}>{userInitials(user) || "?"}</Avatar>;
+}
+
+function initial(name?: string | null) {
+  return typeof name === "string" ? name.charAt(0).toUpperCase() : "";
+}
+
+function userInitials(user: User | Group) {
+  if (user) {
+    return nameInitials(user) || emailInitials(user as User);
```

Comment:
When passing `Group`, `nameInitials(user)` will always return something, so `emailInitials` will guarantee to only render users.

Reply:
Maybe add a type guard like this to prevent casting
```
const isUser = (user: User | Group): user is User => "common_name" in user
```